### PR TITLE
[Android] Add support to TextInput for Image Keyboards

### PR DESCRIPTION
**Pull request by stwiname**
## Summary

Adds [Image Keyboard Support](https://developer.android.com/guide/topics/text/image-keyboard) to `TextInput` on Android. It adds the `onImageInput` prop which calls the callback and provides a local content uri, base64 encoded string of the image, mime type and possibly a url to the image on the web.

## Changelog

[Android] [Added] - `onImageInput` prop to the `TextInput` component

## Test Plan

Tested with Gboard, Fleksy and Tenor gif keyboard on android.

### CODE REVIEW

**Comment by jasonivers** [comment]
There's a pull request that adds support from 3 months ago.  I'm not clear why it isn't merged at this point...
https://github.com/facebook/react-native/pull/26088

**Comment by jasonivers** [comment]
Is there any update on the status of this?  It might not seem very important, but it IS a nice user enhancement, and the pull request is already here (although it appears it needs updated now).  I'd really like to use this, without relying on patching things locally.

**Comment by stwiname** [comment]
@jasonivers, I've rebased so everything is up to date

I don't know what can be done to get someone to merge this

**Comment by roshangm1** [comment]
@cpojer @hramos This is a great PR to be merged if working properly. BTW, what are the ways we can test this locally ?

**Comment by stwiname** [comment]
@roshangm1 are you after example usage or how to run the react-native fork?
Either way here's a simple example

```jsx
<TextInput
  onImageInput={(event) => {
    // Will log an object of type { uri: string, linkUri: string, data: string, mime: string}
    console.log(event.nativeEvent);
  }}
/>
```

Then using an image keyboard on android (Gboard, Fleksy, Tenor gif keyboard and probably a whole lot more) the function will get called. 

On iOS the `onImageInupt` function will never be called

**Comment by jasonivers** [comment]
> @jasonivers, I've rebased so everything is up to date
> 
> I don't know what can be done to get someone to merge this

There appear to be a couple of test failures.  That could be why?

**Comment by stwiname** [comment]
> > @jasonivers, I've rebased so everything is up to date
> > I don't know what can be done to get someone to merge this
> 
> There appear to be a couple of test failures. That could be why?

Those tests seem to be failing on other PRs and unrelated to the changes

**Comment by jasonivers** [comment]
@JoshuaGross - I saw you merged a pull request for something to do with TextInput, and I have no idea how else to find the right person for this PR.  Is there anything that needs done to get the right person's attention?

**Comment by Gustash** [comment]
Bump. This would be an awesome feature

**Comment by Gustash** [comment]
If anyone bumps into this while this PR isn't merged, I've created a package to not only handle this but also images copied to the clipboard on iOS.

https://www.npmjs.com/package/react-native-media-input

**Comment by tsvetann** [comment]
a year and a half later...any luck here?

**Comment by KrisLau** [comment]
Would love to have this merged in as well

File: ReactAndroid/src/main/java/com/facebook/react/views/textinput/ReactEditText.java [hunk 1]
```diff
@@ -212,13 +227,83 @@ protected void onScrollChanged(int horiz, int vert, int oldHoriz, int oldVert) {
...
+    EditorInfoCompat.setContentMimeTypes(outAttrs, new String [] {"image/png",
+      "image/gif",
+      "image/jpg",
+      "image/jpeg",
+      "image/webp"});
+
+    final InputConnectionCompat.OnCommitContentListener callback =
+      new InputConnectionCompat.OnCommitContentListener() {
+        @Override
+        public boolean onCommitContent(InputContentInfoCompat inputContentInfo, int flags, Bundle opts) {
```
Comment (minhnguyen-paidy) [inline ReactAndroid/src/main/java/com/facebook/react/views/textinput/ReactEditText.java:245]: typo here

**Comment by TheMakerOleguch** [comment]
Any update on this?

**Comment by WithakErik** [comment]
Was this ever implemented? I've tried adding `react-native-image-keyboard` (https://www.npmjs.com/package/react-native-image-keyboard) but I still get the error message: `Can't enter this content here` in a `TextInput`

**Comment by shuo-hiwintech** [comment]
> Was this ever implemented? I've tried adding `react-native-image-keyboard` (https://www.npmjs.com/package/react-native-image-keyboard) but I still get the error message: `Can't enter this content here` in a `TextInput`

Same problem. Have you made any progress?

**Pull request closed without merging**